File: src/main/java/exchange/analyzer/model/OrderBookChart.java
package exchange.analyzer.model;

import com.oanda.v20.instrument.OrderBook;
import com.oanda.v20.primitives.DateTime;
import com.oanda.v20.primitives.InstrumentName;

import java.util.HashMap;
import java.util.Map;

public class OrderBookChart {

    private InstrumentName instrumentName;
    private Map<DateTime, OrderBook> orderBookMap;

    public OrderBookChart(InstrumentName instrumentName) {
        this.instrumentName = instrumentName;
        this.orderBookMap = new HashMap<>();
    }

    public void addOrderBook(OrderBook orderBook){
        if (instrumentName.equals(orderBook.getInstrument()))
            orderBookMap.put(orderBook.getTime(), orderBook);
    }

    public InstrumentName getInstrumentName() {
        return instrumentName;
    }

    public Map<DateTime, OrderBook> getOrderBookMap() {
        return orderBookMap;
    }
}

File: front/FxBottle/src/app/model/order-book-chart.ts
import {OrderBook} from "./OrderBook";

export class OrderBookChart {
  private _instrumentName: string;
  private _orderBook: OrderBook;

  constructor(instrumentName: string, orderBook: OrderBook) {
    this._instrumentName = instrumentName;
    this._orderBook = orderBook;
  }

  get instrumentName(): string {
    return this._instrumentName;
  }

  get orderBook(): OrderBook {
    return this._orderBook;
  }
}
